zrml/prediction-markets/src/lib.rs

```diff
@@ -1499,17 +1561,18 @@ mod pallet {
             let market = T::MarketCommons::market(market_id)?;
             if market.status == MarketStatus::Reported {
                 let report = market.report.ok_or(Error::<T>::MarketIsNotReported)?;
-                MarketIdsPerReportBlock::<T>::mutate(&report.at, |ids| {
+                let dispute_duration_ends_at_block =
+                    report.at.saturating_add(market.deadlines.dispute_duration);
+                MarketIdsPerReportBlock::<T>::mutate(&dispute_duration_ends_at_block, |ids| {
                     remove_item::<MarketIdOf<T>, _>(ids, market_id);
                 });
             }
             if market.status == MarketStatus::Disputed {
                 let disputes = Disputes::<T>::get(market_id);
                 if let Some(last_dispute) = disputes.last() {
-                    let at = last_dispute.at;
-                    let mut old_disputes_per_block = MarketIdsPerDisputeBlock::<T>::get(&at);
-                    remove_item::<MarketIdOf<T>, _>(&mut old_disputes_per_block, market_id);
-                    MarketIdsPerDisputeBlock::<T>::mutate(&at, |ids| {
+                    let dispute_duration_ends_at_block =
+                        last_dispute.at.saturating_add(market.deadlines.dispute_duration);
+                    MarketIdsPerDisputeBlock::<T>::mutate(&dispute_duration_ends_at_block, |ids| {
                         remove_item::<MarketIdOf<T>, _>(ids, market_id);
                     });
```

Comment:
What happened there before? I think you corrected it, but maybe there was something other meant with. I couldn't figure it out yet.

Reply:
with this change we are tracking things bit differently in MaketIdsPerDisputeBlock , see https://github.com/zeitgeistpm/zeitgeist/blob/89326d4ecc42214ff888c315f76695a43df1046c/zrml/prediction-markets/src/lib.rs#L986
and this is what is being updated in MigrateMarketIdsPerBlockStorage